lib/improver/wind_gust_diagnostic.py

```diff
@@ -0,0 +1,227 @@
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module containing plugin for WindGustDiagnositic."""
+
+import warnings
+
+import iris
+from iris import FUTURE
+
+from improver.utilities.cube_manipulation import merge_cubes
+
+FUTURE.netcdf_promote = True
+
+
+class WindGustDiagnostic(object):
+
+    """Plugin for calculating wind-gust diagnostic.
+
+    In the model a shear-driven turbulence parameterization is used to
+    estimate wind gusts but in convective situations this can over-estimate the
+    convective gust.
+    This diagnostic takes the Maximum of the values at eachgrid point of
+        a chosen percentile of the wind-gust forecast and
+        a chosen percentile of the wind-speed forecast
+    to produce a better estimate of wind-gust.
+    For example a typical wind-gust could be MAX(gust(50%),windspeed(95%))
+    an extreme wind-gust forecast could be MAX(gust(95%), windspeed(100%))
+
+    See
+    https://github.com/metoppv/improver/files/1244828/WindGustChallenge_v2.pdf
+    for a discussion of the problem and proposed solutions.
+    """
+
+    def __init__(self, percentile_gust, percentile_windspeed):
+        """
+        Create a WindGustDiagnostic plugin for a given set of percentiles.
+
+        Args:
+            percentile_gust: float
+                Percentile value required from wind-gust cube.
+            percentile_windspeed: float
+                Percentile value required from wind-speed cube.
+
+        """
+        self.percentile_gust = percentile_gust
+        self.percentile_windspeed = percentile_windspeed
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = ('<WindGustDiagnostic: wind-gust perc='
+                '{0:3.1f}, wind-speed perc={1:3.1f}>'
+                .format(self.percentile_gust, self.percentile_windspeed))
+        return desc
+
+    def add_metadata(self, cube):
+        """Add metadata to cube for windgust diagnostic.
+
+        Args:
+            cube: iris.cube.Cube instance
+                Cube containing the wind-gust diagnostic data.
+        Returns:
+            result : iris.cube.Cube instance
+                Cube containing the wind-gust diagnostic data with
+                corrected Metadata.
+
+        """
+        result = cube
+        result.standard_name = "wind_speed_of_gust"
+        result.long_name = "wind_gust_diagnostic"
+        if self.percentile_gust == 50.0 and self.percentile_windspeed == 95.0:
+            diagnostic_txt = 'Typical gusts'
+        elif (self.percentile_gust == 95.0 and
+              self.percentile_windspeed == 100.0):
+            diagnostic_txt = 'Extreme gusts'
+        else:
+            diagnostic_txt = str(self)
+        result.attributes.update({'wind_gust_diagnostic': diagnostic_txt})
```

Comment:
I'd much prefer we create some suitable metadata here to store the percentile values as values rather than as part of a string that might have to be passed. That's not to say I hate the idea of Typical gusts and Extreme gusts being used as additional attributes if these terms become an agreed standard that we document with a URI.

Reply:
Again the metadata is not defined in the ticket. I've made this up to match what is describe in the ticket. If there is more appropriate metadata defined somewhere let me know otherwise I see no point in changing this.

Reply:
I do think that we need to come to a joint understanding about what the best temporary metadata conventions are for this issue.

Reply:
I'm not sure I understand the problem with the Metadata. There are two bits

cube.long_name = 'wind_gust_diagnostic'
so that it is obvious that the wind-gust data has been post processed.

cube Attribute
wind_gust_diagnostic which is set to either
  * Extreme gusts
  * Typical gusts
  * A string which describes the plug-in used if a different set of percentiles are used then the ones we prescribe as extreme or typical
This means the gust diagnostic used is uniquely describe but sometime in the future we can modify what we set for extreme or typical (maybe completely rewrite the processing) without having to modify the metadata.
If you have a specific issue with this metadata let me know but in the absence of any other information I think this metadata is adequate for now.

Reply:
Given Nigel's comments I am happy with this for now.